pkg/action/action.go

```diff
@@ -0,0 +1,62 @@
+package action
+
+import (
+	v1api "k8s.io/client-go/pkg/api/v1"
+
+	"github.com/coreos/container-linux-update-operator/pkg/constants"
+)
+
+const (
+	// ActionsUnknown is the zero value for an ActionsResult.
+	// It should not be returned by ActionsCompleted. This value likely
+	// indicates a bug.
+	ActionsUnknown ActionsResult = iota
+	// ActionsPending is returned when there are still actions which haven't
+	// set their configured annotation to true
+	ActionsPending
+	// ActionsSucceeded is returned when all actions have set their configured
+	// annotation to true
+	ActionsSucceeded
+)
+
+// ActionsResult is the result of a given set of actions
+type ActionsResult int
+
+// Action is a pre- or post-reboot action to perform.
+type Action struct {
+	annotation string
+}
+
+// New creates a new action with the given annotation
+func New(annotation string) *Action {
```

Comment:
I thought we were using Action as a placeholder phrase in design docs, not as the actual implementation name. Its rather vague. Especially because CLUO is not actuating the actions, its simply waiting for annotation requirements to be fulfilled.